### ISSUE TITLE
When I use irdac_exporter to obtain Dell information, the log prompts about TLS error information.

### ISSUE THREAD

**Issue by ilanni2460** [mrlhansen/idrac_exporter#41]
When I use irdac_exporter to obtain Dell information, the log prompts about TLS error information.
Oh, by the way, I run it using docker.

idrac_exporter  | 2024-01-08T16:51:58.432 INFO  Server listening on 0.0.0.0:9348
idrac_exporter  | 2024-01-08T16:52:06.610 ERROR Error instantiating metrics collector for host 192.168.73.108: Get "https://192.168.73.108/redfish/v1": tls: server selected unsupported protocol version 301
idrac_exporter  | 2024-01-08T16:52:11.330 ERROR Error instantiating metrics collector for host 192.168.73.107: Get "https://192.168.73.107/redfish/v1": tls: server selected unsupported protocol version 301
idrac_exporter  | 2024-01-08T16:52:21.619 ERROR Error instantiating metrics collector for host 192.168.73.108: Get "https://192.168.73.108/redfish/v1": tls: server selected unsupported protocol version 301

**Comment by mrlhansen** [comment]
Sounds like these machines use an old and by default unsupported version of TLS. I can probably add an option to allow old TLS versions.

**Comment by ilanni2460** [comment]
@mrlhansen 
Thank you very much for your prompt reply.
May I ask which file this is modified in?
If you don't have free time, I can try to modify it first. Thanks again.

**Comment by mrlhansen** [comment]
You can try replacing the `newHttpClient()` function in `internal/collector/client.go` with the following new function (where we explicitly set the minimum allowed TLS version).

```go
func newHttpClient() *http.Client {
	return &http.Client{
		Transport: &http.Transport{
			TLSClientConfig: &tls.Config{
				InsecureSkipVerify: true,
				MinVersion: tls.VersionTLS10,
			},
		},
		Timeout: time.Duration(config.Config.Timeout) * time.Second,
	}
}
```

**Comment by ilanni2460** [comment]
> You can try replacing the `newHttpClient()` function in `internal/collector/client.go` with the following new function (where we explicitly set the minimum allowed TLS version).
> 
> ```go
> func newHttpClient() *http.Client {
> 	return &http.Client{
> 		Transport: &http.Transport{
> 			TLSClientConfig: &tls.Config{
> 				InsecureSkipVerify: true,
> 				MinVersion: tls.VersionTLS10,
> 			},
> 		},
> 		Timeout: time.Duration(config.Config.Timeout) * time.Second,
> 	}
> }
> ```

Okay, thank you very much, I will test it

**Comment by ilanni2460** [comment]
@mrlhansen 

Hi, I used the latest code. When requesting metrics, a new error message was reported, as follows

`Error instantiating metrics collector for host 192.168.73.107: 404 404 Not Found`

![Uploading tls_error_1.png…]()

**Comment by mrlhansen** [comment]
Unfortunately I have zero experience with Huawei machines. Maybe they do not even support the Redfish API - and since they apparently use a deprecated version of TLS, then it's likely that they indeed do not support Redfish. I don't think there is much I can do.

**Comment by ilanni2460** [comment]
@mrlhansen 

I'm very sorry. I made a mistake before. It's not a Huawei machine, but a Dell PowerEdge R620 machine. And after my testing, it can only be opened using the IE browser. Do you support it and how do I do it support Redfish functionality?
Thank you so much……

**Comment by mrlhansen** [comment]
Okay, then in principle it should be possible. This machine is using iDRAC 7 and another user (see #3) did get it working (although, not all information is available via Redfish on these older machines).
* If possible, I would recommend updating iDRAC 7 to the newest possible version (this might also change TLS to a newer version).
* You might need to manually enable Redfish in iDRAC on this machine. On my iDRAC 9 machines this can be done under `iDRAC Settings -> Services` but it might be different on iDRAC 7.

**Comment by ilanni2460** [comment]
Upgrade idrac7 version to
Solved in 2.65.65.65, the tls error message is no longer reported, thank you.

@mrlhansen
